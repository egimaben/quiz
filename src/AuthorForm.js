import React from 'react';
import { MiniDelete } from './quizComponents';
export default class AuthorForm extends React.Component {
    state = {
        authorName: '',
        bookOptions: [],
        authoredBook: '',
        avatar: '',
        booktmp: ''
    };
    handleChange = (event) => {
        this.setState(
            {
                [event.target.name]: event.target.value
            }
        );
    }
    handleSubmit = (event) => {
        event.preventDefault();

        this.props.onSubmit(this.state);
    }
    addBook = () => {
        this.setState(({ bookOptions, booktmp }) => {
            bookOptions.push(booktmp);
            return {
                bookOptions: bookOptions,
                booktmp: ''
            }
        });
    }
    removeBook = (name) => {
        this.setState(({ bookOptions }) => {
            bookOptions.splice(bookOptions.indexOf(name), 1);
            return {
                bookOptions: bookOptions
            }
        });
    }

    render() {
        const { authorName, authoredBook, avatar, booktmp } = this.state;

        return (
            <form onSubmit={this.handleSubmit}>
                <label>
                    Name:
            <br /><input type="text" value={authorName} name='authorName' onChange={this.handleChange} required />
                </label><br />
                <label>
                    Book:
            <br /><input type="text" value={authoredBook} name='authoredBook' onChange={this.handleChange} required />
                </label><br />
                <label>
                    Avatar:
            <br /><input type="text" value={avatar} name='avatar' onChange={this.handleChange} required />
                </label><br />
                <label>
                    Book Options:
            <br />
                    {
                        this.state.bookOptions.map((book,index) => <MiniDelete key={index} item={book} callback={this.removeBook} />)
                    }
                    <br /><input type="text" value={booktmp} name='booktmp' onChange={this.handleChange} />
                    <button className='add-button' type='button' onClick={this.addBook}>add</button>
                </label>
                <br /><input type="submit" value="Submit" />
            </form>
        );
    }
}