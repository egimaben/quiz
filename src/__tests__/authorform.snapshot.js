'use strict'
import React from 'react';
import renderer from 'react-test-renderer';
import AuthorForm from '../components/AuthorForm';
it('displays author form correctly',()=>{
    let onSubmit=jest.fn();
    const tree = renderer.create(
        <AuthorForm onSubmit={onSubmit}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})