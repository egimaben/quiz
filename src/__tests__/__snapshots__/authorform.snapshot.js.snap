// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays author form correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    Name:
    <br />
    <input
      name="authorName"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
  </label>
  <br />
  <label>
    Book:
    <br />
    <input
      name="authoredBook"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
  </label>
  <br />
  <label>
    Avatar:
    <br />
    <input
      name="avatar"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
  </label>
  <br />
  <label>
    Book Options:
    <br />
    <br />
    <input
      name="booktmp"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="add-button"
      onClick={[Function]}
      type="button"
    >
      add
    </button>
  </label>
  <br />
  <input
    type="submit"
    value="Submit"
  />
</form>
`;
